tests/test_integration_layout.py

```diff
@@ -169,3 +162,75 @@ def compose(self) -> ComposeResult:
         assert placeholder.size == expected_placeholders_size
         assert placeholder.styles.offset.x.value == 0.0
         assert app.screen.get_offset(placeholder).x == expected_placeholders_offset_x
+
+
+@pytest.mark.asyncio
+@pytest.mark.integration_test
+@pytest.mark.parametrize(
+    "edge_type,expected_box_inner_size,expected_box_size,expected_top_left_edge_color,expects_visible_char_at_top_left_edge",
+    (
+        # These first 3 types of border edge types are synonyms, and display no borders:
+        ["", Size(SCREEN_W, 1), Size(SCREEN_W, 1), "black", False],
```

Comment:
this test checks the visual result of different border styles - that's pretty much the pytest equivalent of visually checking the layout properties of this:
![Screenshot from 2022-05-18 11-17-09](https://user-images.githubusercontent.com/722388/169023607-a652f64b-4623-482f-82bc-d4200776e442.png)